File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> find = new HashMap<>();
        String key = "";

        for (String s : strs)
        {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            key = new String(chars);

            if (!find.containsKey(key)){
                find.put(key, new ArrayList<>());
            }
            find.get(key).add(s);
        }

        ArrayList<List<String>> answer = new ArrayList<>(find.values());
        return answer;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        check = strs[0]
        lenth = len(check)
        for str in strs:
            while True:
                if check == str:
                    break
                elif check in str:
                    break
                elif len(check) == 0:
                    return ""
                else:
                    check = check[0:lenth-1]
                    lenth -= 1
        return check